### DISCUSSION TITLE
GitHub Copilot could not connect to server. "getaddrinfo ENOTFOUND api.github.com"

### DISCUSSION THREAD

**Discussion by nghiacanhtran** [community/community#7704] [Copilot Conversations]
Hi all
I get big issue with github copilot.I get an message "GitHub Copilot could not connect to server. Extension activation failed: "getaddrinfo ENOTFOUND api.github.com" when install new extension github copilot for vs code

**Comment by davecheney** [comment]
Hello, I'm sorry you are having difficulties. We've found that the copilot icon may not show up if copilot is not able to contact our servers. Are you behind a firewall, vpn, or other technology that may be intercepting your network connections?

**Reply by AlexanderLanin** [reply]
I have the same issue. My setup is:
Local: Windows 10, VPN, local proxy (px) which redirects to corporate proxy. Local proxy configured everywhere (VS Code, http(s)_proxy).
Remote: Ubuntu, local proxy (px) which redirects to corporate proxy. Local proxy configured (http(s)_proxy).

Everything but copilot works fine. Copilot reports:
> GitHub Copilot could not connect to server. Extension activation failed: "getaddrinfo ENOTFOUND api.github.com"

**Reply by Dhamo2005** [reply]
### I have tried this following steps. It works for me.

- Make sure you've connected to internet
- Reopen Or Restart your vs code window
- If these steps are not working then reinstall Copilot extension

**Reply by Gtstride** [reply]
What happened in my case, I needed to open the Github Copilot extension on VScode and I could notice it required me to `upgrade` to the `latest version`. After which my VScode reloaded and it worked just fine. 
Hope this helps someone.

**Comment by TarunavBA** [comment]
So, does anyone has a workaround for this until @github solves this issue?

**Comment by ecyrbe** [comment]
I have the exact same issue when behind a corporate proxy "getaddrinfo ENOTFOUND api.github.com" :
- vs code 1.62
- windows 10

The error is really odd, since dns resolution works fine on the host.

**Reply by N4SIRODDIN3** [reply]
I could resolve this issue by resetting the Network Adapter and the DNS using Command Prompt.
Here you find the detailed steps for this issue:

1. Run as Administrator Command Prompt

1. Type the following commands, pressing enter after finishing each one:

- netsh winsock reset

- ipconfig /release

- netsh int ip reset

- ipconfig /flushdns

- ipconfig /renew

3. Exit Command Prompt

4. Restart your PC.

**Reply by sagoez** [reply]
This worked for me

**Comment by lennartdeknikker** [comment]
I have the same issue, both on my home network as well as on the vpn. Might be happening since I upgraded macOS to Monterey last week. VS code version  1.62.3 (Universal)
 
`[ERROR] [default] [2021-11-24T07:18:19.110Z] GitHub Copilot could not connect to server. Extension activation failed: "getaddrinfo ENOTFOUND api.github.com"`
 
![image](https://user-images.githubusercontent.com/45396678/143201038-aafb5c8f-cbc4-4cc8-9ee7-0deb039cb493.png)

**Reply by xiezhifeng** [reply]
I have same problem, same version of vscode and macOs

**Comment by lennartdeknikker** [comment]
Reinstalling the extension fixed it for me though!

**Reply by wulil0** [reply]
It doesn't work for me

**Reply by BobbySingle** [reply]
Try disable and enable again

**Reply by davecheney** [reply]
Quick suggestion, disabling and reenabling usually involves a restart of vscode. It is likely the restart is resolving the temporary networking issue; not uninstalling/reinstalling the extension.

**Reply by nghiacanhtran** [reply]
> 

It doesn't work for me

**Reply by BruneXX** [reply]
it worked for me, uninstall, reload then install again..

**Reply by miozus** [reply]
It doesn't work for me and I cry for a while 😭

**Comment by jdbruxelles** [comment]
I get the same issue when I lose the internet connexion. I have to restart the workspace/editor to make Copilot work again.

**Comment by riandroraul** [comment]
means most of the problem is internet connection

**Comment by ctsstc** [comment]
I went to try reinstalling the extension and noticed it had a pending update, after the update it started working again.

**Comment by lveillard** [comment]
It happens to me each time after I've used VScode offline.
When coming back to online i have to uninstall copilot and restart VSC
Btw i'm saving a lot of time with Copilot, well done guys!

**Comment by sNok3** [comment]
I had the same issue. I tried to reproduce it and it only happens when an internet connection is absent. In my case restarting VSCode fixed the issue.

**Comment by shaantanu9** [comment]
It happens when you start the VS Code and if there is some issue with the internet in the beginning so for that if you think your net is working fine not just restart the VS Code.
A simple Answer to solve the issue is 
`Restart VS code`
